File: TP/tp-2/versao2/main.c
// Struct Show - C
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>
#include <time.h>
#define MAX_SHOWS 1368      //qt de shows
#define MAX_LINE 500        //qt de char de uma linha

typedef struct {
    char* showId;
    char* type;
    char* title;
    char* director;
    char** cast;
    char* country;
    time_t dateAdded;
    int releaseYear;
    char* rating;
    char* duration;
    char** listedIn;

}Show;

//construtor
Show* criarShow(char* line) {}

int main(void) {

    FILE csvFile = fopen("/home/augustobiza/CCPUC/AED-2/TP/tp-2/tmp/disneyplus.csv", "r");
    if(csvFile == NULL) {
        perror("Error opening file")
    }

    Show* shows[MAX_SHOWS];                                                     //Array de Shows do csv
    int countShow = 0;                                                          //contar o número de Show

    char line[MAX_LINE];
    fgets(line, MAX_LINE, csvFile);                                             //pular o cabeçalho do arquivo

    while(fgets(line, MAX_LINE, csvFile) != NULL && showCount < MAX_SHOWS) {
        line[strcspn(line, "\n")] = '\0';
        shows[showCount++] = criarShow(line);
    }
    fclose(csvFile);

    return 0;
}

File: TP/tp-2/versao2/Main.java
// Classe Show - Java
import java.io.*;
import java.util.*;
import java.time.*;

class Show {

    private String showId;
    private String type;
    private String title;
    private String director;
    private String[] cast;
    private String country;
    private LocalDate dateAdded;
    private int releaseYear;
    private String rating;
    private String duration;
    private String[] listedIn;


    // construtor
    public Show(String line) {}
}


class Main {


    public static void main(String[] args) throws IOException {

        String csvFile = "/home/augustobiza/CCPUC/AED-2/TP/tp-2/tmp/disneyplus.csv";

        Show[] shows = new Show[1368];                                                          //Array de Shows do csv
        int showCount = 0;                                                                      //contar o número de Show

        Scanner scan = new Scanner(new File(csvFile));                                          //scanner vai ler o arquivo
        scan.nextLine();                                                                        //pular o cabeçalho do arquivo
        
        while(scan.hasNextLine()) {                                                             //enquanto tiver linha no arquivo
            shows[showCount++] = new Show(scan.nextLine());                                     //pega a linha e trata ela pra criar um objeto Show e colocar no array shows
        }
        scan.close();                                                                           //fecha o scanner de arquivo


    }
}